doc/dev/osd_internals/manifest.rst

```diff
@@ -46,24 +48,64 @@ There are some rough edges we may want to improve:
     do_osd_ops.  This probably isn't desirable, see Snapshots section
     below for some options on how generally to mix these operations
     with snapshots.
+    --> Modify SET_CHUNK to set user_modify = false not to trigger a new clone.
   * Appears to assume that the corresponding section of the object
     does not exist (sets FLAG_MISSING) but does not check whether the
     corresponding extent exists already in the object.
+    --> Set FLAG_DIRTY if the corresponding section of the object exists.
+
   * Appears to clear the manifest unconditionally if not chunked,
     that's probably wrong.  We should return an error if it's a
     REDIRECT.
+    --> add the following lines ::
+
+	case CEPH_OSD_OP_SET_CHUNK:
+	  if (oi.manifest.is_redirect()) {
+	    result = -EINVAL;
+	    goto fail;
+	  }
+
 
 * TIER_PROMOTE:
 
   * Document control flow in this file.  The CHUNKED case is
     particularly unintuitive and could use some explanation.
+    --> 
+    Based on the chunk_map in a head object (each chunk's state can be 
+    dirty or missing), chunked objects for the head object 
+    can be retrieved when tier_promote is invoked.
+    After promotion is completed, all chunks in the object should be clean state.
+
+    One thing to note is snapshotted manifest object.
+    Suppose that there are 10(9), 5(4). If we want to read clone id 5,
+    need to read chunks according to the chunk_map for clone id 5.
+    But, because the base tier keeps object_info_t in the case of manifest object,
+    the read with snapshot can work.
+    If clone id 5 is dirty or clean, the chunks will be copied from the base tier.
```

Comment:
I'm not sure what you mean by this.  Do you mean that if the referenced chunks exist in clone id 5 you'll read them from there directly?  Note: promoting to, flushing, and evicting portions of clones is currently impossible -> you'll have to add support for that.

Reply:
I mean, if the clone is either dirty or clean state, we don't need to copy chunks from the lower tier, we can read it from the base tier directly.

Also, as you mentioned above
> Fundamentally, I think we need to be able to manipulate the manifest
> status of clones because we want to be able to dynamically promote,
> flush (if the state was dirty when the clone was created), and evict
> clones.
> As such, I think we want to make a new op type which permits writes on
> clones.
I think we need a new op type to support promoting, flushing, and evicting clone if I understand your intention correctly. So, what I mean is that we can support such features if
we add a new op.

Reply:
I'm actually unsure of whether you need a new op, you might just have to add a clone_id field to the existing ones.  However it needs to be done is fine though.